.github/workflows/setup-tofu.yml

```diff
@@ -0,0 +1,314 @@
+
+    - name: Setup OpenTofu - ${{ matrix['tofu-versions'] }}
+      uses: ./
+      with:
+        tofu_version: ${{ matrix['tofu-versions'] }}
+
+    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+      if: ${{ matrix['tofu-versions'] != 'latest' }}
+      run: tofu version | grep ${{ matrix['tofu-versions']}}
+
+    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+      if: ${{ matrix['tofu-versions'] == 'latest' }}
+      run: tofu version | grep 'OpenTofu v'
+
+  tofu-versions-no-wrapper:
+    name: 'OpenTofu Versions No Wrapper'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        # TODO: add more versions to test when the semver feature is implemented
+        tofu-versions: [latest]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+
+    - name: Setup OpenTofu (no wrapper) - ${{ matrix['tofu-versions'] }}
+      uses: ./
+      with:
+        tofu_version: ${{ matrix['tofu-versions'] }}
+        tofu_wrapper: false
+
+    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+      if: ${{ matrix['tofu-versions'] != 'latest' }}
+      run: tofu version | grep ${{ matrix['tofu-versions']}}
+
+    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+      if: ${{ matrix['tofu-versions'] == 'latest' }}
+      run: tofu version | grep 'OpenTofu v'
+
+# TODO: uncomment when the semver feature is implemented
+#  tofu-versions-constraints:
+#    name: 'OpenTofu Versions Constraints'
+#    runs-on: ${{ matrix.os }}
+#    strategy:
+#      matrix:
+#        os: [ubuntu-latest, windows-latest, macos-latest]
+#        tofu-versions: [~0.12, 0.12.x, <0.13.0]
+#    steps:
+#    - name: Checkout
+#      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+#
+#    - name: Setup OpenTofu - ${{ matrix['tofu-versions'] }}
+#      uses: ./
+#      with:
+#        tofu_version: ${{ matrix['tofu-versions'] }}
+#
+#    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+#      run: tofu version | grep 'OpenTofu v0\.12'
+
+# TODO: uncomment when the semver feature is implemented
+#  tofu-versions-constraints-no-wrapper:
+#    name: 'OpenTofu Versions Constraints No Wrapper'
+#    runs-on: ${{ matrix.os }}
+#    strategy:
+#      matrix:
+#        os: [ubuntu-latest, windows-latest, macos-latest]
+#        tofu-versions: [~0.12, 0.12.x, <0.13.0]
+#    steps:
+#    - name: Checkout
+#      uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
+#
+#    - name: Setup OpenTofu (no wrapper) - ${{ matrix['tofu-versions'] }}
+#      uses: ./
+#      with:
+#        tofu_version: ${{ matrix['tofu-versions'] }}
+#        tofu_wrapper: false
+#
+#    - name: Validate OpenTofu Version - ${{ matrix['tofu-versions'] }}
+#      run: tofu version | grep 'OpenTofu v0\.12'
+
+  tofu-credentials-cloud:
```

Comment:
why do we need TF cloud?

Reply:
Please clarify the requirements then. My understanding was, that we want to provide github action where users would seamlessly swap terraform with tofu. So if users were to use terraform cloud, they should still be able to do so regardless of cli, shouldn't they?

Reply:
You are correct AFAIK

Reply:
Would you mind me keeping this job as part of this CI pipeline then?

Reply:
Yes, let's keep it as a whole for nwo